src/scraper/scraper.cpp

```diff
@@ -4925,6 +4925,41 @@ ScraperStatsAndVerifiedBeacons GetScraperStatsAndVerifiedBeacons(const Converged
     return stats_and_verified_beacons;
 }
 
+// This is for rpc report functions.
+ScraperPendingBeaconMap GetPendingBeaconsForReport()
+{
+    return GetConsensusBeaconList().mPendingMap;
```

Comment:
If we use the consensus list, none of the RPCs will show pending beacons advertised within the last hour. Those are available from `NN::GetBeaconRegistry().PendingBeacons()` directly. Maybe we need `beaconreport`, `pendingbeaconreport`, and `beaconconvergence` functions, where the last provides introspection into the scraper's data structures. What do you think?

I know I said that I find separate RPCs more useful, but it seems to make more sense to group the last one together after thinking about it in this way.

Reply:
So what I think you are saying is taking the current method of the pendingbeaconreport, and put it in the verifiedbeaconreport and rename that beaconconvergence, then replace the guts of the pendingbeaconreport with a direct access to .PendingBeacons. The only problem with that approach is I don't think the .PendingBeacons has the max depth cutoff, but maybe that is really not an issue.

Reply:
Yeah, there's no max-depth cutoff for `BeaconRegistry::Beacons()` or `BeaconRegistry::PendingBeacons()`. This matches the old `AppCache` for which the RPCs just displayed everything in memory. Useful for debugging to see how state mutates or lingers, but we can add a check for `Expired()` as in `GetConsensusBeaconList()` if you think that makes sense.

Reply:
I think that isn't necessary.